webapp/src/main/java/org/apache/atlas/web/security/AtlasKeycloakAuthenticationProvider.java

```diff
@@ -65,10 +75,31 @@ public Authentication authenticate(Authentication authentication) {
         authentication = new KeycloakAuthenticationToken(token.getAccount(), token.isInteractive(), grantedAuthorities);
       }
     }
+    if(authentication.getName().startsWith("service-account-apikey")) {
```

Comment:
@ektavarma10 In the final PR, please add a comment here explaining the "why" around this. Also, the code section isn't very readable. Small things like adding appropriate newlines and spaces (1 space after `try`, 1 space after `if`, etc.)